docker/resources/common-scripts/configure

```diff
@@ -0,0 +1,147 @@
+#!/usr/bin/env bash
+# Licensed to the Apache Software Foundation (ASF) under one or more
+# contributor license agreements.  See the NOTICE file distributed with
+# this work for additional information regarding copyright ownership.
+# The ASF licenses this file to You under the Apache License, Version 2.0
+# (the "License"); you may not use this file except in compliance with
+# the License.  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# --- for broker
+
+# If KAFKA_PROCESS_ROLES is defined it means we are running in KRaft mode
+
+# unset KAFKA_ADVERTISED_LISTENERS from ENV in KRaft mode when running as controller only
+if [[ -n "${KAFKA_PROCESS_ROLES-}" ]]
+then
+  echo "Running in KRaft mode..."
+  ub ensure CLUSTER_ID
+  if [[ $KAFKA_PROCESS_ROLES == "controller" ]]
+  then
+    if [[ -n "${KAFKA_ADVERTISED_LISTENERS-}" ]]
+    then
+      echo "KAFKA_ADVERTISED_LISTENERS is not supported on a KRaft controller."
+      exit 1
+    else
+      unset KAFKA_ADVERTISED_LISTENERS
+    fi
+  else
+    ub ensure KAFKA_ADVERTISED_LISTENERS
+  fi 
+else
+  echo "Running in Zookeeper mode..."
+  ub ensure KAFKA_ZOOKEEPER_CONNECT
+  ub ensure KAFKA_ADVERTISED_LISTENERS
+fi
+
+# By default, LISTENERS is derived from ADVERTISED_LISTENERS by replacing
```

Comment:
As per [docs](https://kafka.apache.org/documentation/#brokerconfigs_advertised.listeners), ADVERTISED_LISTENERS is derived from LISTENERS so this seems counter intuitive for someone used to Kafka's configs. 
Also ADVERTISED_LISTENERS is optional and we are making it mandatory in above if block
```ub ensure KAFKA_ADVERTISED_LISTENERS```

Reply:
In case of docker image usage will mostly be external to the docker network, hence it makes sense to keep ADVERTISED_LISTENERS as mandatory. 
Although it might be a bit counter intuitive, but it makes sense in the context of docker image to define ADVERTISED_LISTENERS and derive LISTENERS because of the same reason.